File: spaced_learning_app/lib/features/review_history/domain/entities/skip_reason.dart
import 'package:freezed_annotation/freezed_annotation.dart';

@JsonEnum(unknownEnumValue: SkipReason.unknown)
enum SkipReason {
  @JsonValue('forgot') forgot,
  @JsonValue('busy') busy,
  @JsonValue('other') other,
  unknown,
}

File: spaced-learning-api/src/main/java/com/spacedlearning/entity/enums/SkipReason.java
package com.spacedlearning.entity.enums;

/**
 * Enum representing the reason for skipping a review
 */
public enum SkipReason {
    FORGOT("forgot"),
    BUSY("busy"),
    OTHER("other");

    private final String value;

    SkipReason(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    @Override
    public String toString() {
        return value;
    }
}
